scripts/generate-ebpf-types.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+# Copyright Antimetal, Inc. All rights reserved.
+#
+# Use of this source code is governed by a source available license that can be found in the
+# LICENSE file or at:
+# https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd)"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+NC='\033[0m' # No Color
+
+# Check if we're in the right directory
+if [[ ! -f "${PROJECT_ROOT}/go.mod" ]]; then
+    echo -e "${RED}Error: This script must be run from the system-agent repository root${NC}"
+    exit 1
+fi
+
+echo -e "${GREEN}Generating eBPF type definitions...${NC}"
+
+# Build the generator tool
+echo "Building ebpf-typegen..."
+if ! go build -o "${PROJECT_ROOT}/bin/ebpf-typegen" "${PROJECT_ROOT}/cmd/ebpf-typegen/main.go"; then
+    echo -e "${RED}Failed to build ebpf-typegen${NC}"
+    exit 1
+fi
+
+# Find all *_types.h files in ebpf/include
+HEADER_FILES=$(find "${PROJECT_ROOT}/ebpf/include" -name "*_types.h" -type f)
+
+if [[ -z "$HEADER_FILES" ]]; then
+    echo -e "${YELLOW}No *_types.h files found in ebpf/include${NC}"
+    exit 0
+fi
+
+# Generate Go types for each header file
+for header in $HEADER_FILES; do
+    basename=$(basename "$header" .h)
+    collector_name="${basename%_types}"
+    
+    output_file="${PROJECT_ROOT}/pkg/performance/collectors/${basename}.go"
+    
+    echo "Generating $output_file from $header..."
+    
+    if ! "${PROJECT_ROOT}/bin/ebpf-typegen" \
+        -input "$header" \
+        -output "$output_file" \
+        -package collectors; then
+        echo -e "${RED}Failed to generate types for $header${NC}"
+        exit 1
+    fi
+done
+
+# Run go fmt on generated files
+echo "Formatting generated files..."
+go fmt "${PROJECT_ROOT}/pkg/performance/collectors/"*_types.go
```

Comment:
nit: no need to worry about this atm but the generated go files should already be go fmt formatted.